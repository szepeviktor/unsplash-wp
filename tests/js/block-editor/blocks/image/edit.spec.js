/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { cloneDeep } from 'lodash';

/**
 * WordPress dependencies
 */
import { registerStore } from '@wordpress/data';

/**
 * Internal dependencies
 */
import Edit from '../../../../../assets/src/block-editor/blocks/image/edit';

// Mock the <InspectorControls> component only, so that the other components in this package behave as usual.
jest.mock( '@wordpress/block-editor', () => {
	const original = require.requireActual( '@wordpress/block-editor' );
	return {
		...original,
		InspectorControls: ( { children } ) => children,
	};
} );

registerStore( 'core', {
	reducer: jest.fn(),
	selectors: {
		getMedia: () => {
			return {
				id: 2,
				media_details: {
					width: 1000,
					height: 1000,
					file: 'example-photo.png',
					sizes: {
						medium: {
							file: 'example-photo-300x240.png',
							width: 300,
							height: 240,
							mime_type: 'image/png',
							source_url:
								'https://images.unsplash.com/example-photo-300x240.png',
						},
						thumbnail: {
							file: 'example-photo-150x150.png',
							width: 150,
							height: 150,
							mime_type: 'image/png',
							source_url:
								'https://images.unsplash.com/example-photo-150x150.png',
						},
						full: {
							file: 'example-photo.png',
							width: 587,
							height: 469,
							mime_type: 'image/png',
							source_url: 'https://images.unsplash.com/example-photo',
						},
					},
				},
			};
		},
	},
} );

const baseProps = {
	attributes: {
		url: 'https://images.unsplash.com/example-photo',
		alt: 'Example Photo',
		caption:
			'Photo by <a href="https://unsplash.com/@user" rel="nofollow">User</a> on <a href="https://unsplash.com" rel="nofollow">Unsplash</a>',
		align: 'wide',
		id: 2,
		href: 'https://unsplash.com',
		linkClass: 'url-class',
		linkDestination: 'attachment',
		sizeSlug: 'large',
	},
	setAttributes: jest.fn(),
	className: 'is-style-default',
	isSelected: true,
};

const setup = props => {
	return render( <Edit { ...props } /> );
};

describe( 'blocks: unspash/image: edit', () => {
	function Unsplash() {
		return {
			open: jest.fn(),
			on: jest.fn(),
		};
	}

	beforeAll( () => {
		global.wp = {
			media: {
				view: {
					MediaFrame: {
						Unsplash,
					},
				},
			},
		};
	} );

	it( 'matches snapshot when no image is selected', () => {
		const props = cloneDeep( baseProps );
		props.attributes.url = '';
		props.attributes.id = '';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when an image is selected', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
