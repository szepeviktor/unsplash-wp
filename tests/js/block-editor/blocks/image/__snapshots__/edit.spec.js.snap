// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: unspash/image: edit matches snapshot when an image is selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <figure
        class="is-style-default is-focused size-large alignwide"
      >
        <div>
          <div
            class="components-panel__body is-opened"
          >
            <h2
              class="components-panel__body-title"
            >
              <button
                aria-expanded="true"
                class="components-button components-panel__body-toggle"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  <svg
                    aria-hidden="true"
                    class="components-panel__arrow"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12,8l-6,6l1.41,1.41L12,10.83l4.59,4.58L18,14L12,8z"
                    />
                  </svg>
                </span>
                Image settings
              </button>
            </h2>
            <div
              class="components-base-control"
            >
              <div
                class="components-base-control__field"
              >
                <label
                  class="components-base-control__label"
                  for="inspector-textarea-control-0"
                >
                  Alt text (alternative text)
                </label>
                <textarea
                  aria-describedby="inspector-textarea-control-0__help"
                  class="components-textarea-control__input"
                  id="inspector-textarea-control-0"
                  rows="4"
                >
                  Example Photo
                </textarea>
              </div>
              <p
                class="components-base-control__help"
                id="inspector-textarea-control-0__help"
              >
                <a
                  class="components-external-link"
                  href="https://www.w3.org/WAI/tutorials/images/decision-tree"
                  rel="external noreferrer noopener"
                  target="_blank"
                >
                  Describe the purpose of the image
                  <span
                    class="components-visually-hidden"
                  >
                    (opens in a new tab)
                  </span>
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-external components-external-link__icon"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 3h8v8l-2-1V6.92l-5.6 5.59-1.41-1.41L14.08 5H10zm3 12v-3l2-2v7H3V6h8L9 8H5v7h7z"
                    />
                  </svg>
                </a>
                Leave empty if the image is purely decorative.
              </p>
            </div>
            <div
              class="components-base-control"
            >
              <div
                class="components-base-control__field"
              >
                <label
                  class="components-base-control__label"
                  for="inspector-select-control-0"
                >
                  Image size
                </label>
                <select
                  class="components-select-control__input"
                  id="inspector-select-control-0"
                >
                  <option
                    value="thumbnail"
                  >
                    Thumbnail
                  </option>
                  <option
                    value="medium"
                  >
                    Medium
                  </option>
                  <option
                    value="full"
                  >
                    Full Size
                  </option>
                </select>
              </div>
            </div>
          </div>
          <img
            alt="Example Photo"
            src="https://images.unsplash.com/example-photo"
          />
        </div>
        <figcaption
          aria-label="Write captionâ€¦"
          aria-multiline="true"
          class="rich-text block-editor-rich-text__editable blocks-gallery-caption"
          contenteditable="true"
          role="textbox"
          style="white-space: pre-wrap;"
        >
          Photo by 
          <a
            href="https://unsplash.com/@user"
            rel="nofollow"
          >
            User
          </a>
           on 
          <a
            href="https://unsplash.com"
            rel="nofollow"
          >
            Unsplash
          </a>
          
        </figcaption>
      </figure>
    </div>
  </body>,
  "container": <div>
    <figure
      class="is-style-default is-focused size-large alignwide"
    >
      <div>
        <div
          class="components-panel__body is-opened"
        >
          <h2
            class="components-panel__body-title"
          >
            <button
              aria-expanded="true"
              class="components-button components-panel__body-toggle"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                <svg
                  aria-hidden="true"
                  class="components-panel__arrow"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12,8l-6,6l1.41,1.41L12,10.83l4.59,4.58L18,14L12,8z"
                  />
                </svg>
              </span>
              Image settings
            </button>
          </h2>
          <div
            class="components-base-control"
          >
            <div
              class="components-base-control__field"
            >
              <label
                class="components-base-control__label"
                for="inspector-textarea-control-0"
              >
                Alt text (alternative text)
              </label>
              <textarea
                aria-describedby="inspector-textarea-control-0__help"
                class="components-textarea-control__input"
                id="inspector-textarea-control-0"
                rows="4"
              >
                Example Photo
              </textarea>
            </div>
            <p
              class="components-base-control__help"
              id="inspector-textarea-control-0__help"
            >
              <a
                class="components-external-link"
                href="https://www.w3.org/WAI/tutorials/images/decision-tree"
                rel="external noreferrer noopener"
                target="_blank"
              >
                Describe the purpose of the image
                <span
                  class="components-visually-hidden"
                >
                  (opens in a new tab)
                </span>
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-external components-external-link__icon"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 3h8v8l-2-1V6.92l-5.6 5.59-1.41-1.41L14.08 5H10zm3 12v-3l2-2v7H3V6h8L9 8H5v7h7z"
                  />
                </svg>
              </a>
              Leave empty if the image is purely decorative.
            </p>
          </div>
          <div
            class="components-base-control"
          >
            <div
              class="components-base-control__field"
            >
              <label
                class="components-base-control__label"
                for="inspector-select-control-0"
              >
                Image size
              </label>
              <select
                class="components-select-control__input"
                id="inspector-select-control-0"
              >
                <option
                  value="thumbnail"
                >
                  Thumbnail
                </option>
                <option
                  value="medium"
                >
                  Medium
                </option>
                <option
                  value="full"
                >
                  Full Size
                </option>
              </select>
            </div>
          </div>
        </div>
        <img
          alt="Example Photo"
          src="https://images.unsplash.com/example-photo"
        />
      </div>
      <figcaption
        aria-label="Write captionâ€¦"
        aria-multiline="true"
        class="rich-text block-editor-rich-text__editable blocks-gallery-caption"
        contenteditable="true"
        role="textbox"
        style="white-space: pre-wrap;"
      >
        Photo by 
        <a
          href="https://unsplash.com/@user"
          rel="nofollow"
        >
          User
        </a>
         on 
        <a
          href="https://unsplash.com"
          rel="nofollow"
        >
          Unsplash
        </a>
        
      </figcaption>
    </figure>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`blocks: unspash/image: edit matches snapshot when no image is selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="components-placeholder placeholderClassName is-small"
      >
        <iframe
          aria-hidden="true"
          aria-label="resize-listener"
          frameborder="0"
          src="about:blank"
          style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
          tabindex="-1"
        />
        <div
          class="components-placeholder__label"
        >
          <svg
            height="18"
            version="1.1"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="surface1"
            >
              <path
                d="M 7.5 6.75 L 7.5 0 L 16.5 0 L 16.5 6.75 Z M 16.5 10.5 L 24 10.5 L 24 24 L 0 24 L 0 10.5 L 7.5 10.5 L 7.5 17.25 L 16.5 17.25 Z M 16.5 10.5 "
              />
            </g>
          </svg>
          Unsplash
        </div>
        <div
          class="components-placeholder__instructions"
        >
          Search and select from the internet's source of freely usable images
        </div>
        <div
          class="components-placeholder__fieldset"
        >
          <button
            class="components-button is-primary"
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="components-placeholder placeholderClassName is-small"
    >
      <iframe
        aria-hidden="true"
        aria-label="resize-listener"
        frameborder="0"
        src="about:blank"
        style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
        tabindex="-1"
      />
      <div
        class="components-placeholder__label"
      >
        <svg
          height="18"
          version="1.1"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="surface1"
          >
            <path
              d="M 7.5 6.75 L 7.5 0 L 16.5 0 L 16.5 6.75 Z M 16.5 10.5 L 24 10.5 L 24 24 L 0 24 L 0 10.5 L 7.5 10.5 L 7.5 17.25 L 16.5 17.25 Z M 16.5 10.5 "
            />
          </g>
        </svg>
        Unsplash
      </div>
      <div
        class="components-placeholder__instructions"
      >
        Search and select from the internet's source of freely usable images
      </div>
      <div
        class="components-placeholder__fieldset"
      >
        <button
          class="components-button is-primary"
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
